tests/PHPStan/Analyser/data/bug-3387.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Analyser\Bug2851;
+
+use function PHPStan\Analyser\assertType;
+
+function (array $input) {
+	$results = [];
+	foreach ($input as $value) {
+		$results[] = $value;
+
+		assertType('array<int, mixed>', $results);
+		assertType('int', count($results));
+	}
+};
+
+function (array $input) {
+	$results = [];
+	foreach ($input as $value) {
+		array_unshift($results, $value);
+
+		assertType('array<int, mixed>&nonEmpty', $results);
+		assertType('int', count($results));
+	}
+};
+
+function (array $input, string $key) {
+	$results = [$key => []];
+	foreach ($input as $value) {
+		$results[$key][] = $value;
+
+		assertType('array<int, mixed>', $results[$key]);
+		assertType('int', count($results[$key]));
+	}
+};
+
+function (array $input, string $key) {
+	$results = [$key => []];
+	foreach ($input as $value) {
+		array_push($results[$key], $value);
+
+		assertType('mixed', $results[$key]);
```

Comment:
I'm not 100% sure if `mixed` is expected. I haven't figured out how to do, but should it be `array<int, mixed>`?